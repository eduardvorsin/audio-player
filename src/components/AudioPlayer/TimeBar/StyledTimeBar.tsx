import styled from "styled-components";

interface StyledTimeBarProps {
	className?: string,
};

export const StyledTimeBar = styled.time<StyledTimeBarProps>`
	font-size: 16px;
	letter-spacing: 0.3px;
	margin-bottom: 10px;
	align-self: flex-end;
	color: var(--secondary-color);
`;