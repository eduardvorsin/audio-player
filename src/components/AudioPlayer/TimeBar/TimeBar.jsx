import React from "react";
import PropTypes from 'prop-types';
import { StyledTimeBar } from "./StyledTimeBar";

export const TimeBar = ({
  currentTime,
  duration,
  className,
  ...props
}) => {
  return (
    <StyledTimeBar
      className={className}
      data-testid='time-bar'
      {...props}
    >
      {currentTime}/{duration}
    </StyledTimeBar>
  );
};


TimeBar.propTypes = {
  currentTime: PropTypes.string,
  duration: PropTypes.string,
}

TimeBar.defaultProps = {
  currentTime: '00:00',
  duration: '00:00',
}
