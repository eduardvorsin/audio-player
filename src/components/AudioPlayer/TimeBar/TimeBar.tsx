import React from "react";
import { StyledTimeBar } from "./StyledTimeBar";

export interface TimeBarProps extends Partial<React.TimeHTMLAttributes<HTMLTimeElement>> {
  className?: string,
  currentTime?: string,
  duration?: string,
};

export const TimeBar: React.FC<TimeBarProps> = ({
  currentTime = '00:00',
  duration = '00:00',
  className = '',
  ...props
}) => {
  return (
    <StyledTimeBar
      className={className}
      data-testid='time-bar'
      {...props}
    >
      {currentTime}/{duration}
    </StyledTimeBar>
  );
};
