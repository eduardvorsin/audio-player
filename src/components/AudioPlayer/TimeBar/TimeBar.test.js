import { render, screen } from "@testing-library/react";
import { TimeBar } from "./TimeBar";

describe('TimeBar component tests', () => {
  test('renders correctly', () => {
    render(
      <TimeBar
        currentTime={'2:43'}
        duration={'5:21'}
      />
    );

    expect(screen.getByText('2:43/5:21')).toBeInTheDocument();
  });

  test('renders without passed props', () => {
    render(<TimeBar />);

    expect(screen.getByText('00:00/00:00')).toBeInTheDocument();
  });

  test('snapshot with default prop values', () => {
    render(
      <TimeBar />
    );

    expect(screen.getByTestId('time-bar')).toMatchSnapshot();
  });

  test('snapshot with all the passed props', () => {
    render(
      <TimeBar
        duration='10:30'
        currentTime='5:26'
        className='test'
      />
    );

    expect(screen.getByTestId('time-bar')).toMatchSnapshot();
  });
});
