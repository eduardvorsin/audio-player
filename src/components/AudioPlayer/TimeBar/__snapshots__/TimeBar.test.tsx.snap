// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeBar component tests snapshot with all the passed props 1`] = `
<time
  class="sc-bcXHqe dFrouj test"
  data-testid="time-bar"
>
  5:26
  /
  10:30
</time>
`;

exports[`TimeBar component tests snapshot with default prop values 1`] = `
<time
  class="sc-bcXHqe dFrouj"
  data-testid="time-bar"
>
  00:00
  /
  00:00
</time>
`;
